pkg/controllers/management/authprovisioningv2/machineconfigs.go

```diff
@@ -0,0 +1,79 @@
+package authprovisioningv2
+
+import (
+	"strings"
+
+	v32 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/controllers/management/nodetemplate"
+	"github.com/rancher/rancher/pkg/controllers/management/rbac"
+	"github.com/rancher/wrangler/pkg/data"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+func validMachineConfigGVK(gvk schema.GroupVersionKind) bool {
+	return gvk.Group == "rke-machine-config.cattle.io" &&
```

Comment:
Will the schema or gvk ever be null here? If so, add a null check

Reply:
No. In Go, only pointer types can be `nil`.